networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/peers/Eth2PeerManager.java

```diff
@@ -105,7 +111,49 @@ public static Eth2PeerManager create(
         peerValidatorFactory,
         attestationSubnetService,
         rpcEncoding,
-        eth2RpcPingInterval);
+        eth2RpcPingInterval,
+        eth2StatusUpdateInterval);
+  }
+
+  private void setUpPeriodicTasksForPeer(Eth2Peer peer) {
+    SafeFuture<Void> periodicStatusUpdateTask = periodicallyUpdatePeerStatus(peer);
+    SafeFuture<Void> periodicPingTask = periodicallyPingPeer(peer);
+    peer.subscribeDisconnect(() -> periodicStatusUpdateTask.cancel(false));
+    peer.subscribeDisconnect(() -> periodicPingTask.cancel(false));
+  }
+
+  @SuppressWarnings("FutureReturnValueIgnored")
+  SafeFuture<Void> periodicallyUpdatePeerStatus(Eth2Peer peer) {
+    return asyncRunner
+        .runWithFixedDelay(
+            peer::sendStatus,
+            eth2StatusUpdateInterval.getSeconds(),
+            TimeUnit.SECONDS,
+            (err) -> LOG.debug("Exception updating peer status.", err))
+        .exceptionallyCompose(
```

Comment:
This variant of `runWithFixedDelay` returns `Future` which is actually never completes (until `cancel()` call). So the only method which makes sense here is `Future.cancel()`. Any exception happened in the periodic task is reported via exceptionHandler (`(err) -> LOG.debug("Exception updating peer status.", err))`)

That's probably my fault and the `runWithFixedDelay` method should return something like `Cancellable` instead of `SafeFuture<Void>`

Reply:
We discussed this further offline, and Anton made a patch that switched `runWithFixedDelay` to return a `Cancellable` object instead of a `SafeFuture`. I've included the patch in this PR.